timely/src/progress/subgraph.rs

```diff
@@ -530,26 +530,37 @@ where
         assert_eq!(self.children[0].outputs, self.inputs());
         assert_eq!(self.children[0].inputs, self.outputs());
 
-        let mut internal_summary = Vec::with_capacity(self.inputs());
-        for summary in self.scope_summary.iter() {
-            let scope_summary = summary.iter()
-                .map(|output| {
-                    output.elements()
-                        .iter()
-                        .cloned()
-                        .fold(
-                            Antichain::with_capacity(output.elements().len()),
-                            |mut antichain, path_summary| {
-                                antichain.insert(TInner::summarize(path_summary));
-                                antichain
-                            },
-                        )
-                })
-                .collect();
-
-            internal_summary.push(scope_summary);
+        // Note that we need to have `self.inputs()` elements in the summary
+        // with each element containing `self.outputs()` antichains regardless
+        // of how long `self.scope_summary` is
+        let mut internal_summary: Vec<Vec<_>> = (0..self.inputs())
+            .map(|_| (0..self.outputs()).map(|_| Antichain::new()).collect())
+            .collect();
```

Comment:
Do you prefer this to 
```rust
let mut internal_summary = vec![vec![Antichain::new(); self.outputs()]; self.inputs()];
```
or was it just a matter that it was easier to type the above vs recover the prior code?

Reply:
The above pre-allocates (and initializes) the proper number of elements so that no further reallocations have to be made

Reply:
I think this does too. The `vec![thing; len]` desugars down to roughly the same code with a `thing.clone()` instead of `Antichain::new()`.